src/methods/join.js

```diff
@@ -411,13 +422,15 @@ export class JoinImplementation {
     joinCondition: AbstractValue,
     m1: PropertyBindings,
     m2: PropertyBindings,
-    c1: CreatedObjects,
-    c2: CreatedObjects
+    co1: CreatedObjects,
+    co2: CreatedObjects,
+    ca1: CreatedAbstracts,
```

Comment:
You never use ca1 and ca2, so why pass them in?

Reply:
So we could use them at a later point if needed. I'm happy if we remove them though.

In terms of performance, this change added no noticeable overhead in my testing and I frequently compare compile times between commits.

Reply:
I'd actually prefer it if this entire PR is reverted. I think membership of CreatedAbstracts is too coarse a measure to be useful in a truly effective heuristic for inlining vs outlining decisions. We should spend some more time discussing and designing this feature before making invasive code changes. I don't know if this PR is performance neutral or not, but the heap usage doubled in the period when it went in. It seems worthwhile to revert it, if only to establish that it is not the cause of the regression.

Reply:
@hermanventer Sure, if you feel that's the best course then I'm happy for it to be reverted.